be/src/storage/lake/tablet_writer.h

```diff
@@ -108,12 +124,20 @@ class TabletWriter {
     // allow to set custom tablet schema for writer, used in partial update
     void set_tablet_schema(TabletSchemaCSPtr schema) { _schema = std::move(schema); }
 
+protected:
+    // the pattern of segment_path is like this xxx/table_id/partition_id/data/segment_file_name
+    void set_segment_size(std::string segment_path, uint64_t segment_size) {
+        auto pos = segment_path.find_last_of("/");
+        _files_to_size.emplace(segment_path.substr(pos + 1), segment_size);
+    }
+
 protected:
     TabletManager* _tablet_mgr;
     int64_t _tablet_id;
     TabletSchemaCSPtr _schema;
     int64_t _txn_id;
     std::vector<std::string> _files;
```

Comment:
can we change _files to `vector<pair<string, uint64>` or `vector<FileInfo>`?

Reply:
The reason is for file and file_size are inserted in different places, so  we need a map to find the object by fileName